src/util/FileMonitor.h

```diff
@@ -37,38 +36,36 @@ namespace lsst::qserv::util {
 /// events wait for.
 class FileMonitor {
 public:
-	using Ptr = std::shared_ptr<FileMonitor>;
+    using Ptr = std::shared_ptr<FileMonitor>;
 
-	FileMonitor() = delete;
-	FileMonitor(std::string const& fileName) : _fileName(fileName) {
-		_setup();
-	}
+    FileMonitor() = delete;
+    FileMonitor(std::string const& fileName) : _fileName(fileName) { _setup(); }
 
-	/// Stops and joins _thrd, if joinable.
-	~FileMonitor();
+    /// Stops and joins _thrd, if joinable.
+    ~FileMonitor();
 
-	FileMonitor(FileMonitor const&) = delete;
-	FileMonitor& operator=(FileMonitor const&) = delete;
+    FileMonitor(FileMonitor const&) = delete;
+    FileMonitor& operator=(FileMonitor const&) = delete;
 
-	/// start running `_thrd` containing `checkLoop()`.
-	void run();
+    /// start running `_thrd` containing `checkLoop()`.
+    void run();
 
-	/// Stop the thread `_thrd`.
-	void stop() { _loop = false; }
+    /// Stop the thread `_thrd`.
+    void stop() { _loop = false; }
 
-	/// Join to `_thrd`, if it is joinable.
-	void join();
+    /// Join to `_thrd`, if it is joinable.
+    void join();
```

Comment:
You don't need to have this method because it has no users but the class's destructor.

Reply:
Just because it isn't really needed, doesn't mean that it shouldn't be there. It helps make the code more readable to have these functions for basic tasks separate.

Reply:
My point was that there is no reason to have this method in the `public:` area of the class. Seeing a class with to many publicly visible external controls is confusing. And it complicates the interface of the class. It also makes the design fragile as these methods could be called not when the original developer expects it.